pkg/apis/logging/v1/clusterlogging_types.go

```diff
@@ -150,6 +150,9 @@ type ElasticsearchSpec struct {
 
 	// +optional
 	RedundancyPolicy elasticsearch.RedundancyPolicyType `json:"redundancyPolicy"`
+
+	// Specification of the Kibana Proxy component
+	ProxySpec `json:"proxy,omitempty"`
```

Comment:
This reuses unfortunately the same proxy spec from kibana. The proxy component refers for kibana to `oauth-proxy`, but for elasticsearch to `elasticsearch-proxy`. The two ProxySpecs should be different structs. In addition please align the comment block for this field to match the purpose.

Reply:
i disagree, i think it is acceptable to reuse this spec given its just:
```
type ProxySpec struct {
	// +nullable
	Resources *v1.ResourceRequirements `json:"resources"`
}
```

i think it is unnecessary to create another struct that represents this same thing but just for elasticsearch's proxy

Reply:
What if we want to extend this struct with kibana's oauth-proxy only fields?

Reply:
we can add fields specific to oauth at that time

Reply:
But the struct will be a union of fields for oauth-proxy and fields for both proxies. Isn't this awkward?

Reply:
i dont think so, its for a proxy... it's delineated by function. similar to logstore or collection or visualization or forwarder